Makefile

```diff
@@ -1,16 +1,20 @@
-.PHONY: all
-all: protos
+ROOT = $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+BUILD_DIR = $(ROOT)/out
+CHANNEL ?= canary
+VERSION ?= $(shell git describe --tags --dirty  --always)
+GOOS ?= $(shell ./bin/go version | awk '{print $$NF}' | cut -d/ -f1)
+GOARCH ?= $(shell ./bin/go version | awk '{print $$NF}' | cut -d/ -f2)
+BIN = $(BUILD_DIR)/pranadb-$(GOOS)-$(GOARCH)
 
-.PHONY: protos
-protos:
-	make -C protos
+.PHONY: all build test docker-image
 
-.PHONY: clean
-clean:
-	make -C protos clean
-	go clean -testcache
+all: build protos
```

Comment:
Should the `protos` prerequisite be before `build`?

Reply:
ah, yeah, good catch.